lib/rucio/daemons/conveyor/poller.py

```diff
@@ -74,29 +72,28 @@ def poller(once=False,
     except NoOptionError:
         timeout = None
 
-    logging.info('poller starting - process (%i/%i) thread (%i/%i) bulk (%i) timeout (%s)' % (process, total_processes,
-                                                                                              thread, total_threads,
-                                                                                              db_bulk, timeout))
-
     executable = ' '.join(sys.argv)
     hostname = socket.getfqdn()
     pid = os.getpid()
     hb_thread = threading.current_thread()
     heartbeat.sanity_check(executable=executable, hostname=hostname)
-    hb = heartbeat.live(executable, hostname, pid, hb_thread)
+    heart_beat = heartbeat.live(executable, hostname, pid, hb_thread)
+    prepend_str = 'Thread [%i/%i] : ' % (heart_beat['assign_thread'] + 1, heart_beat['nr_threads'])
+    logging.info(prepend_str + 'Poller starting - db_bulk (%i) fts_bulk (%i) timeout (%s)' % (db_bulk, fts_bulk, timeout))
+
+    time.sleep(10)  # To prevent running on the same partition if all the poller restart at the same time
```

Comment:
but they will all wait 10 seconds? shouldn't it include a random time delay instead?

Reply:
Yes, they all wait 10 seconds (the time that all the threads start and report their heartbeat). I'm not sure what we gain with a random time sleep.

Reply:
Yes this pause is just done once so that the heartbeat done in line 80 is done by all threads and when they read it again in line 85 they all know about each other.